primefaces/src/main/java/org/primefaces/model/filter/InFilterConstraint.java

```diff
@@ -49,6 +50,11 @@ else if (filter instanceof Collection) {
             if (Objects.equals(value, o)) {
                 return true;
             }
+
+            // GitHub #8106 check for "" comparison
+            if (o instanceof String && o.toString().length() == 0 && value == null) {
```

Comment:
can o.toString return null?
maybe better user LangUtils.isEmpty(o.toString())

Reply:
Yep switched to Langutils!  Good Catch